database/models/pharmacist.model.ts

```diff
@@ -0,0 +1,74 @@
+import mongoose, { Types } from "mongoose";
+
+const pharmacistSchema = new mongoose.Schema({
+    licenseNumber: {
+        type: String,
+        required: [true, 'license number is required'],
+        unique: true,
+        maxLength: 50
+    },
+    username: {
+        type: String,
+        required: [true, 'Username is required'],
+        unique: true,
+        minLength: [3, 'Username is too short'],
+        maxLength: 50
+    },
+    email: {
+        type: String,
+        required: [true, 'Email is required'],
+        unique: true,
+        lowercase: true,
+        trim: true
+    },
+    password: {
+        type: String,
+        required: [true, 'Password is required'],
+        minLength: [6, 'Password must be at least 6 characters long']
+    },
+    pharmacy: {
+        type: Types.ObjectId,
+        ref: "Pharmacy",
+        required: true
+    },
+    inventory: {
+        type: Types.ObjectId,
+        ref: "inventory",
+        required: true
+    },
+    role: {
```

Comment:
role: in the top (first)

role: {
            type: String,
            default: 'pharmacist'
        }